Sources/Logger/OSLoggerProvider.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  OSLoggerProvider.swift
+//  CrossmintSDK
+//
+//  Created by Tomas Martins on 2/12/25.
+//
+
+import Foundation
+import OSLog
+
+public class OSLoggerProvider: LoggerProvider {
+    private let osLogger: OSLog
+    private let subsystem: String
+
+    public init(category: String) {
+        self.subsystem = "CrossmintSDK"
+        self.osLogger = OSLog(subsystem: subsystem, category: category)
+    }
+
+    func debug(_ message: String, attributes: [String: any Encodable]?) {
+        os_log(.debug, log: osLogger, "%{public}@", message)
+    }
+
+    func error(_ message: String, attributes: [String: any Encodable]?) {
+        os_log(.error, log: osLogger, "%{public}@", message)
+    }
+
+    func info(_ message: String, attributes: [String: any Encodable]?) {
+        os_log(.info, log: osLogger, "%{public}@", message)
+    }
+
+    func warn(_ message: String, attributes: [String: any Encodable]?) {
+        os_log(.default, log: osLogger, "%{public}@", message)
+    }
```

Comment:
in both this class and `DataDogLoggerProvider`, we have these 4 functions that are exactly the same. What do you think about having `log(_ level, _ message: String, attributes:)` on `LoggerProvider` instead?